test/InternalFunctionsTest.php

```diff
@@ -9,16 +9,53 @@
 final class InternalFunctionsTest extends TestCase
 {
     /**
-     * Tests that the JSONB `@?` operator is not substituted by the parameter parser.
+     * Tests that unnamed parameters are substituted correctly.
+     *
+     * @see parseNamedParams
+     * @dataProvider provideUnnamedParams
+     */
+    public function testParseUnnamedParams(string $sqlIn, string $sqlOut): void
+    {
+        self::assertSame($sqlOut, parseNamedParams($sqlIn, $dummy));
+    }
+
+    public function provideUnnamedParams(): iterable
+    {
+        return [
+            'Bare' => ['SELECT ?', 'SELECT $1'],
+            'Parenthesized' => ['SELECT (?)', 'SELECT ($1)'],
+            'Row constructor' => ['SELECT (?, ?)', 'SELECT ($1, $2)'],
+        ];
+    }
+
+    /**
+     * Tests that operators containing the question mark character, that could be confused with unnamed parameters,
+     * are parsed correctly.
      *
      * @see parseNamedParams
      * @see https://github.com/amphp/postgres/issues/39
+     * @dataProvider provideProblematicOperators
      */
-    public function testParseJsonbOperator(): void
+    public function testParseProblematicOperators(string $sql): void
```

Comment:
I think the test cases can be merged into one test method, as the problematic ones shouldn't be problematic after the fix.

Reply:
For what benefit? Two distinct sets is clearer.

Reply:
Better overview, after all it's one test-set that needs to work.

Reply:
I cannot see the benefit of combining test cases that perform substitutions with test cases that do not. They are complete opposites. As such it would be unconscionable for me to make such a change, but obviously you can still make whatever changes you want after merging since it is your project.